action.yml

```diff
@@ -0,0 +1,76 @@
+name: 'Generate and Commit Badge to Wiki'
+description: 'Generates a status badge using shields.io and commits it to the repository wiki.'
+author: 'Darma-tasking'
+
+# Define the inputs the action will accept
+inputs:
+  names:
+    description: 'The labels for the badges (e.g., matrix item).'
+    required: true
+  results:
+    description: 'The results of build step (success, failure or cancelled).'
+    required: true
+  github_token:
+    description: 'GitHub token with permissions to write to the wiki.'
+    required: true
+
+# Define the execution logic for the action
+runs:
+  using: 'composite'
+  steps:
+    - name: Clone Wiki Repository
+      shell: bash
+      run: |
+        git clone https://x-access-token:${{ inputs.github_token }}@github.com/DARMA-tasking/badge-generator.wiki.git wiki
+
+    - name: Generate and Save Badge
+      shell: bash
+      run: |
+        set -euo pipefail
```

Comment:
@JacobDomagala For the future: this script is long enough to put in an .sh file. Should be easier to get shellcheck feedback this way too.